src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2340,36 +2327,20 @@ internal override int MetadataArity
                 }
             }
 
-            private readonly PENamedTypeSymbolNonGeneric _underlying;
-            internal override bool IsNativeInt { get; }
-
             internal override NamedTypeSymbol AsNativeInt(bool asNativeInt)
             {
                 Debug.Assert(this.SpecialType == SpecialType.System_IntPtr || this.SpecialType == SpecialType.System_UIntPtr);
 
-                if (IsNativeInt == asNativeInt)
-                {
-                    return this;
-                }
-
-                return asNativeInt ? new PENamedTypeSymbolNonGeneric(this, isNativeInt: true) : _underlying;
+                return asNativeInt ?
+                    (NamedTypeSymbol)new NativeIntegerTypeSymbol(this) :
```

Comment:
>(NamedTypeSymbol)new NativeIntegerTypeSymbol(this) [](start = 20, length = 50)

It is probably worth caching this symbol on a containing assembly

Reply:
Added PROTOTYPE comment.

---
In reply to: [370300427](https://github.com/dotnet/roslyn/pull/40994#discussion_r370300427) [](ancestors = 370300427)